src/regdesk/states/unpaid_result_state.js

```diff
@@ -0,0 +1,121 @@
+import { BadgeLabelBuilder } from '../label_builder.js';
+import { RegState } from './reg_state.js';
+
+/**
+ * Loading state
+ */
+export class UnpaidResultState extends RegState {
+  /**
+   * Get the transition events this state can process.
+   */
+  static get events() {
+    return {
+      CANCEL: 'CANCEL',
+      PAID: 'PAID',
+    };
+  }
+
+  /** @type {BadgeLabelBuilder} */
+  #labelBuilder;
+
+  /**
+   * Initializes a new instance of the SingleResultState class.
+   * @param {RegMachineArgs} regMachineArgs - Standard argument set for states.
+   * @param {*} eventMap - Mapping of this object's transition events to new states.
+   */
+  constructor(regMachineArgs, eventMap) {
+    super(regMachineArgs, eventMap);
+
+    this.regUnpaidNotAccepting = this.screenRow.querySelector('#regUnpaidNotAccepting');
+
+    this.regPreferredName = this.screenRow.querySelector('#regPreferredNameUnpaid');
+    this.regLegalName = this.screenRow.querySelector('#regLegalNameUnpaid');
+    this.regBirthdate = this.screenRow.querySelector('#regBirthdateUnpaid');
+    this.regAge = this.screenRow.querySelector('#regAgeUnpaid');
+    this.regPayment = this.screenRow.querySelector('#regPaymentUnpaid');
+
+    this.badgeCanvas = this.screenRow.querySelector('#regBadgePreviewUnpaid');
+
+    this.regPaidBtn = this.screenRow.querySelector('#regMarkPaidBtn');
+    this.regPaidBtn.addEventListener('click', this.markRegistrantAsPaid.bind(this));
+
+    this.regPaidBtnText = this.regPaidBtn.querySelector('#regMarkPaidBtnText');
+    this.regPaidBtnSpinner = this.regPaidBtn.querySelector('#regMarkPaidBtnSpinner');
+    this.regPaidBtnTextSpinner = this.regPaidBtn.querySelector('#regMarkPaidBtnTextSpinner');
+  }
+
+  /**
+   * Enter this state.
+   */
+  enterState() {
+    this.show(this.screenRow);
+    this.printButton.invisible();
+    this.cancelButton.visible()
+      .setTransitionCallback(this, this.constructor.events.CANCEL);
+
+    const acceptingPayments = this.commManager.acceptingPayments;
+    const reg = this.commManager.selectedSearchResult;
+
+    this.regPreferredName.value = reg.preferredName;
+    this.regLegalName.value = reg.legalName;
+    this.regBirthdate.value = reg.birthdate;
+    const age = this.#getAge(reg.birthdate);
+    this.regAge.textContent = age;
+    this.regPayment.value = reg.amountDue;
+
+    // TODO: Move this to some Registrant class?
```

Comment:
Could make the argument that the payment screen shouldn't have anything to do with the badge screen. IE, just remove this whole portion.

Reply:
I don't disagree, but I do want to keep the badge preview on the payment screen as it's a quick and useful additional check for "This is actually you, right?" that the cashiers need to do. I don't remember if it was FC but I remember a story of two people with the same name and same birthday getting mixed up at registration. Probably the corner case of corner cases, but slapping the badge canvas up there makes it ezpz to disambiguate.

~That and I'm really proud of that code and want to throw it everywhere :V~